plutus-tx-plugin/test/Plugin/Errors/Spec.hs

```diff
@@ -53,7 +53,7 @@ negativeInt :: CompiledCode Integer
 negativeInt = plc (Proxy @"negativeInt") (-1 :: Integer)
 
 caseInt :: CompiledCode (Integer -> Bool)
-caseInt = plc (Proxy @"caseInt") (\(i::Integer) -> case i of { S# _ -> True; _ -> False; } )
+caseInt = plc (Proxy @"caseInt") (\(i::Integer) -> case i of { IS _ -> True; _ -> False; } )
```

Comment:
Reason for this change: (1) `S#` is deprecated in favor of `IS`. (2) In GHC 8.10.7, the `CoreExpr` we get for `caseInt` is

```
\ (ds_dmy9 :: Integer) ->
  case ds_dmy9 of {
    __DEFAULT -> False;
    S# _ -> True
  }
```

and so the plugin terminates with the desired error message "Error: Unsupported feature: Cannot case on a value on type Integer".

But in GHC 9.2.4, if we still use `S#`, what we get is
```
\ (ds_dehf :: Integer) ->
  $mS#
    @LiftedRep
    @Bool
    ds_dehf
    (\ _ -> True)
    (\ _ -> False)
```

and the error message becomes "Unsupported feature: Kind: GHC.Types.RuntimeRep".

This is because in GHC 8.10.7, `S#` is a data constructor of `Integer`. In GHC 9.2.4, the data constructor is `IS`, and `S#` is a pattern synonym for `IS`. Pattern synonyms are [compiled into levity-polymorphic CPS functions](https://gitlab.haskell.org/ghc/ghc/-/blob/a54827e0b48af33fa9cfde6ad131c6751c2fe321/compiler/GHC/Core/PatSyn.hs#L258-274).